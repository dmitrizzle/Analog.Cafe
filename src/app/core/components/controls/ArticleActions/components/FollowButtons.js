import React from "react"
import styled, { css } from "styled-components"

import { GA } from "../../../../../utils"
import Etsy from "../../../icons/Social/components/Etsy"
import Feedly from "../../../icons/Social/components/Feedly"
import Instagram from "../../../icons/Social/components/Instagram"
import Itunes from "../../../icons/Social/components/Itunes"
import LinkButton from "../../Button/components/LinkButton"
import Twitter from "../../../icons/Social/components/Twitter"

export const FollowButtons = styled.div`
  text-align: center;
  background: ${props => props.theme.color.background()};
  padding: 0.5em 0 0;
  box-shadow: 0 1px 1px rgba(44, 44, 44, 0.25),
    0 0 0 1px rgba(44, 44, 44, 0.125);

  svg {
    height: 2em;
  }
`
export const FollowButton = css`
  padding: 0;
  box-shadow: none;
  margin: 0;
  display: inline-block;
  background: transparent;
  width: 3em;
  height: 2em;
  &:active {
    background: #fff !important;
  }
  svg path {
    fill: #999 !important;
    stroke: transparent;
  }
`
const FeedlyButton = styled(LinkButton)`
  ${FollowButton};
`
const TwitterButton = styled(LinkButton)`
  ${FollowButton} svg {
    height: 2.65em;
    margin-top: -0.3em;
  }
`
const InstagramButton = styled(LinkButton)`
  ${FollowButton} svg {
    height: 1.4em;
    margin: -0.4em 0 0 0.2em;
  }
`
const ItunesButton = styled(LinkButton)`
  ${FollowButton} svg {
    height: 1.4em;
    margin: -0.4em 0 0 0.2em;
  }
`
const EtsyButton = styled(LinkButton)`
  ${FollowButton} svg {
    height: 1.4em;
    margin: -0.4em 0 0 0.2em;
  }
`

// const FacebookButton = styled(LinkButton)`
//   ${FollowButton} svg {
//     height: 1.4em;
//     margin: -0.4em 0 0 0.2em;
//     background: #999 !important;
//     border-radius: 0.25em;
//     path {
//       fill: #fff !important;
//     }
//   }
// `;

export default props => {
  return (
    <FollowButtons>
      <EtsyButton
        to="https://www.etsy.com/ca/shop/AnalogCafeShop"
        onClick={() => {
          GA.event({
            category: "Campaign",
            action: "FollowButtons.follow_etsy"
          })
        }}
      >
        <Etsy />
      </EtsyButton>
      <FeedlyButton
        to="http://bit.ly/FeedAnalog"
        onClick={() => {
          GA.event({
            category: "Campaign",
            action: "FollowButtons.follow_feedly"
          })
        }}
      >
        <Feedly />
      </FeedlyButton>
      <TwitterButton
        to="https://twitter.com/analog_cafe"
        onClick={() => {
          GA.event({
            category: "Campaign",
            action: "FollowButtons.follow_twitter"
          })
        }}
      >
        <Twitter />
      </TwitterButton>
      <InstagramButton
        to="https://instagram.com/analog_cafe"
        onClick={() => {
          GA.event({
            category: "Campaign",
            action: "FollowButtons.follow_instagram"
          })
        }}
      >
        <Instagram />
      </InstagramButton>
      <ItunesButton
        to="https://itunes.apple.com/us/podcast/analog-cafe-a-film-photography-podcast/id1452093436?mt=2&app=podcast"
        onClick={() => {
          GA.event({
            category: "Campaign",
            action: "FollowButtons.follow_itunes"
          })
        }}
      >
        <Itunes />
      </ItunesButton>
    </FollowButtons>
  )
}
