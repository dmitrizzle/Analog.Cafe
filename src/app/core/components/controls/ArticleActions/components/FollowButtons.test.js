import { shallow } from "enzyme"
import React from "react"

import FollowButtons from "./FollowButtons"

it("Render FollowButtons without crashing", () => {
  shallow(<FollowButtons />)
})
