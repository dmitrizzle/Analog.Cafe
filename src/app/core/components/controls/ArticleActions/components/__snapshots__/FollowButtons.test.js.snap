// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render FollowButtons without crashing, matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Styled(Component)
          onClick={[Function]}
          to="http://bit.ly/FeedAnalog"
>
          <Unknown />
</Styled(Component)>,
        <Styled(Component)
          onClick={[Function]}
          to="https://twitter.com/analog_cafe"
>
          <Unknown />
</Styled(Component)>,
        <Styled(Component)
          onClick={[Function]}
          to="https://instagram.com/analog_cafe"
>
          <Unknown />
</Styled(Component)>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Unknown />,
          "onClick": [Function],
          "to": "http://bit.ly/FeedAnalog",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Unknown />,
          "onClick": [Function],
          "to": "https://twitter.com/analog_cafe",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Unknown />,
          "onClick": [Function],
          "to": "https://instagram.com/analog_cafe",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Styled(Component)
            onClick={[Function]}
            to="http://bit.ly/FeedAnalog"
>
            <Unknown />
</Styled(Component)>,
          <Styled(Component)
            onClick={[Function]}
            to="https://twitter.com/analog_cafe"
>
            <Unknown />
</Styled(Component)>,
          <Styled(Component)
            onClick={[Function]}
            to="https://instagram.com/analog_cafe"
>
            <Unknown />
</Styled(Component)>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Unknown />,
            "onClick": [Function],
            "to": "http://bit.ly/FeedAnalog",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Unknown />,
            "onClick": [Function],
            "to": "https://twitter.com/analog_cafe",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Unknown />,
            "onClick": [Function],
            "to": "https://instagram.com/analog_cafe",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
